config/sampler_config.go

```diff
@@ -45,12 +47,379 @@ type RulesBasedSamplerCondition struct {
 	Field    string
 	Operator string
 	Value    interface{}
+	Datatype string
+	Matches  func(value any, exists bool) bool
+}
+
+func (r *RulesBasedSamplerCondition) Init() error {
+	return r.setMatchesFunction()
 }
 
 func (r *RulesBasedSamplerCondition) String() string {
 	return fmt.Sprintf("%+v", *r)
 }
 
+func (r *RulesBasedSamplerCondition) setMatchesFunction() error {
+	switch r.Operator {
+	case "exists":
```

Comment:
nit: The strings for these operators now live here and in https://github.com/honeycombio/refinery/blob/e7f9bf393fc884e0549c11c1ea73bd30249910c6/sample/rules.go#L195. Would be nice to start using constants/enums

Reply:
I meant to say `also` not `now.  Sorry about that.

Reply:
Tyler is right here -- it would be worthwhile to define some constants like:

```
const (
     EQ = "="
     NE = "!="
     LT = "<"
...
```

This helps to avoid silly but common copy-paste errors.

Reply:
covered via https://github.com/honeycombio/refinery/issues/592